consumer/event_consumer.py

```diff
@@ -28,13 +28,17 @@ def connect(self):
         self.qconnector.connect()
 
     def start_consuming(self):
-        def stream_consumer(routing_key, payload):
+        def stream_consumer(routing_key, payload, properties=None, delivery_tag=None):
             payload_dict = json.loads(payload)
 
-            event: BaseEvent = self.event_cls()
-            event.from_dict(payload_dict)
+            if self.event_cls:
```

Comment:
This is irrelevant for pg events. If you want to use the same consumer you can use the same Base Event structure as well